app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -280,8 +317,54 @@ public void onClick(View v) {
         });
         //Show outdoor map on start
         showOutdoorMap();
+        relativeButtonThread();
+
+    }
+
+    /**
+     * Positions the Locate Me button as well as the
+     * Directions button relative to the bottomsheet at all times
+     */
+    private void relativeButtonThread(){
+        //Thread
+        Thread thread = new Thread(){
+            @Override
+            public void run(){
+                while(true) {
+                    try {
+                        sleep(20);
```

Comment:
It is 20 milliseconds as Sam wrote. But, I find this value to be too low, because 20 milliseconds means that 50 times a second this thread is polling to check for the position. I find this to be too resource intensive. Set it to 200-300 milliseconds, which means that 5-3.333 times a second, this thread is doing the check. Remember that the sheet animates, so we don't need the button to be super responsive and move instantaneously, play with the values, but I would not be surprised if even a 600 millisecond delay gave satisfactory results. It will consume less resources.